lib/EventListener.php

```diff
@@ -222,18 +222,44 @@ protected function getUsersForShares(array $shares, string $owner): array {
 			}
 		}
 
+		$groupAddedUsers = false;
 		if (!empty($groups)) {
 			foreach ($groups as $gid) {
 				$group = $this->groupManager->get($gid);
 				if ($group instanceof IGroup) {
 					foreach ($group->getUsers() as $user) {
+						$groupAddedUsers = true;
 						$users[] = $user->getUID();
 					}
 				}
 			}
 		}
 
-		return array_unique($users);
+		$users = array_unique($users);
+
+		if (!$groupAddedUsers) {
+			return $users;
+		}
+
+		/** @var \OCA\DAV\CalDAV\Sharing\Service $service */
+		$service = \OCP\Server::get(\OCA\DAV\CalDAV\Sharing\Service::class);
+		if (!method_exists($service, 'getUnshares')) {
+			return $users;
+		}
```

Comment:
Is this needed if you intend to bump server version to 30+?

Reply:
If I bump, then not.